baseapp/baseapp.go

```diff
@@ -606,6 +616,21 @@ func (app *BaseApp) cacheTxContext(ctx sdk.Context, txBytes []byte) (sdk.Context
 	return ctx.WithMultiStore(msCache), msCache
 }
 
+func (app *BaseApp) SetCometInfo(ci coreinfo.CometInfo) {
+	app.cometInfo = ci
+
+}
+
+func (app *BaseApp) GetBlockInfo(ctx context.Context) coreinfo.BlockInfo {
+	sdkCtx := sdk.UnwrapSDKContext(ctx)
+
+	return NewBlockInfo(sdkCtx.BlockHeight(), sdkCtx.HeaderHash(), sdkCtx.BlockTime(), sdkCtx.BlockHeader().ChainID)
```

Comment:
If we're making `BlockInfo` + `CometInfo` interfaces then `sdk.Context` can just implement `BlockInfo` and `CometInfo` and we don't have to do any copying. Or we make `BlockInfo` a struct as I'm suggesting

Reply:
but dont we want this to be passed to modules, putting it on sdkContext is allowing modules to do the same as now.

Reply:
modules wouldn't get sdk.Context, they would get the interface and sdk.Context would just implement it. but like I said before, a struct may be better here negating this point